model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/RealmAdapter.java

```diff
@@ -860,19 +862,38 @@ public void setSmtpConfig(Map<String, String> smtpConfig) {
 
     @Override
     public Stream<IdentityProviderModel> getIdentityProvidersStream() {
-        if (isUpdated()) return updated.getIdentityProvidersStream();
-        return cached.getIdentityProviders().stream();
+        if (isUpdated()) return updated.getIdentityProvidersStream().map(this::orgsEnabledAwareIdPModel);
+        return cached.getIdentityProviders().stream().map(this::orgsEnabledAwareIdPModel);
     }
 
     @Override
     public IdentityProviderModel getIdentityProviderByAlias(String alias) {
-        if (isUpdated()) return updated.getIdentityProviderByAlias(alias);
+        if (isUpdated()) return orgsEnabledAwareIdPModel(updated.getIdentityProviderByAlias(alias));
         return getIdentityProvidersStream()
                 .filter(model -> Objects.equals(model.getAlias(), alias))
                 .findFirst()
+                .map(this::orgsEnabledAwareIdPModel)
                 .orElse(null);
     }
 
+    private IdentityProviderModel orgsEnabledAwareIdPModel(IdentityProviderModel idp) {
+        if (!Profile.isFeatureEnabled(Profile.Feature.ORGANIZATION)) return idp;
+        return new IdentityProviderModel(idp) {
+            @Override
+            public boolean isEnabled() {
+                // if IdP is bound to an org
+                if (getOrganizationId() != null) {
+                    // for export we need to export actual isEnabled value
+                    boolean isExport = session.getAttributeOrDefault("export", Boolean.FALSE);
```

Comment:
Thinking more about this, perhaps we can skip this for now and wait for feedback once we start working on the export/import task.

The fact that importing the realm back where those brokers will be disabled so that the admin has to manually enable them (and the org feature/flag) is acceptable, I would say. Just trying to focus on the main goal and handle other (not blockers) things in a proper time. 

wdyt?

Reply:
I've checked exporting users and since the workflow there also requires enabling users one by one, I've reverted this change and let's wait for the feedback as you suggest.